app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,103 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to tasks_path
+      return
+    end
+
+    if @task.completed
+      @status = "completed"
```

Comment:
this is a bit of a "things to look forward to on larger projects," but these strings kind of come out of nowhere, even though they should feel significant. On larger projects, the best practice would be to keep these strings as constant variables somewhere that Rails could reach them (not necessarily in the controller). Keeping it as a constant somewhere allows for consistency (so you won't have to remember in other pieces of code that use this, things like "is it capitalized, how do you spell it" etc.... (also for translation purposes!!!))